### PR TITLE
Feature/search bar

### PR DESCRIPTION
**Pull request by salesfelipe**
#### What is the purpose of this pull request?
Add the search-bar component

#### What problem is this solving?
Now the user is able to search for a specific product in the catalog of products.

#### How should this be manually tested?

[workspace](https://search-bar--storecomponents.myvtex.com/)

#### Screenshots or example usage

image

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: packages/search-bar/src/components/ResultsList.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+const listClassNames =
+  'vtex-results__list w-100 mt3 border-box bw1 br2 b--solid outline-0 near-black b--light-gray hover-b--silver bg-white f5 pv4 ph6'
+const listItemClassNames =
+  'vtex-results__item w-95 mt1 pa1 border-box bw1 br2 flex b--solid outline-0 near-black b--light-gray hover-b--silver bg-white f5 pv4 ph6'
+
+function getImageUrl(image) {
+  return (image.match(/http:(.*?)"/g) || [''])[0]
+}
+
+function getLinkText(href, criteria) {
```
Comment (andreldsa) [inline packages/search-bar/src/components/ResultsList.js]: Use the GraphQL slug directly.

File: packages/search-bar/src/index.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+import Downshift from 'downshift'
+
+export default class SearchBar extends Component {
+  render() {
+    const { placeholder } = this.props
+
+    return (
+      <div className="vtex-searchbar">
+        <Downshift onChange={item => console.log(item)}>
```
Comment (andreldsa) [inline packages/search-bar/src/index.js]: console?

File: packages/search-bar/src/queries/autocomplete.gql [hunk 3]
```diff
@@ -0,0 +1,10 @@
+query Autocomplete($maxRows: Int, $inputValue: String) {
+  autocomplete(maxRows: $maxRows, searchTerm: $inputValue) {
+    itemsReturned {
+      thumb
+      name
+      href
```
Comment (andreldsa) [inline packages/search-bar/src/queries/autocomplete.gql]: add the slug property

File: packages/search-bar/src/components/ResultsList.js (hunk 1 above)
Comment (salesfelipe) [reply packages/search-bar/src/components/ResultsList.js]: 👍

File: packages/search-bar/src/index.js (hunk 2 above)
Comment (salesfelipe) [reply packages/search-bar/src/index.js]: 👍

File: packages/search-bar/src/queries/autocomplete.gql (hunk 3 above)
Comment (salesfelipe) [reply packages/search-bar/src/queries/autocomplete.gql]: 👍

File: packages/search-bar/package.json [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  "files": [
+    "dist/"
+  ],
+  "peerDependencies": {
+    "prop-types": "^15.6.1",
+    "react": "^16.3.2",
+    "react-dom": "^16.3.2"
+  },
+  "dependencies": {
+    "@vtex/styleguide": "2.0.0-rc.34",
```
Comment (andreldsa) [inline packages/search-bar/package.json]: Vtex styleguide can already be used importing directly inside de code, without npm dependencies.

File: react/components/search-bar/src/components/AutocompleteInput.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+export default class AutocompleteInput extends Component {
+  render() {
+    return (
+      <Input size="large" onKeyPress={this.props.onKeyDown} {...this.props} />
+    )
+  }
+}
+
+AutocompleteInput.propTypes = {
+  autoComplete: PropTypes.string.isRequired,
```
Comment (brunojdo) [inline react/components/search-bar/src/components/AutocompleteInput.js]: docs

File: react/components/search-bar/src/components/ResultsList.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+            </div>
+          </Link>
+        ))}
+      </div>
+    )
+  }
+}
+
+const itemProps = PropTypes.shape({
+  thumb: PropTypes.string,
```
Comment (brunojdo) [inline react/components/search-bar/src/components/ResultsList.js]: doc

File: react/components/search-bar/src/components/ResultsList.js [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+})
+
+ResultsList.propTypes = {
+  /** Graphql data response. */
+  data: PropTypes.shape({
+    autocomplete: PropTypes.shape({
+      itemsReturned: PropTypes.arrayOf(itemProps),
+    }),
+    loading: PropTypes.bool.isRequired,
+  }),
```
Comment (brunojdo) [inline react/components/search-bar/src/components/ResultsList.js]: docs

Comment (salesfelipe) [reply react/components/search-bar/src/components/ResultsList.js]: 👍

**Pull request merged**